### PR TITLE
Ui multi tenancy - add in view services configuration and operational start and stop

### PR DESCRIPTION
**Pull request by davidradl**
After this change the ui config contains a list of view services, similar to the way the OMAG Server config contains a list of access services.  The view services register them selves and are configured using the same pattern as the access services.

The configuration and start and stop of the server does not currently effect the runtime. This is required prerequisite work so that the runtime can can the view service instance handlers to obtain their configuration.

### CODE REVIEW

File: open-metadata-implementation/access-services/discovery-engine/discovery-engine-server/src/main/java/org/odpi/openmetadata/accessservices/discoveryengine/handlers/DiscoveryConfigurationHandler.java [hunk 1]
```diff
@@ -283,11 +283,11 @@ public  DiscoveryEngineProperties getDiscoveryEngineByName(String    userId,
         invalidParameterHandler.validateUserId(userId, methodName);
         invalidParameterHandler.validatePaging(startingFrom, maximumResults, methodName);
 
-        List<EntityDetail> retrievedEntities = repositoryHandler.getEntitiesByType(userId,
-                                                                                   DiscoveryEnginePropertiesMapper.DISCOVERY_ENGINE_TYPE_GUID,
-                                                                                   startingFrom,
-                                                                                   maximumResults,
-                                                                                   methodName);
+        List<EntityDetail> retrievedEntities = repositoryHandler.getEntityByType(userId,
+                                                                                 DiscoveryEnginePropertiesMapper.DISCOVERY_ENGINE_TYPE_GUID,
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/discovery-engine/discovery-engine-server/src/main/java/org/odpi/openmetadata/accessservices/discoveryengine/handlers/DiscoveryConfigurationHandler.java]: This seems to be restoring the old name of the method

**Comment by davidradl** [comment]
@mandy-chessell on "Why are these files being deleted?"; I am not sure which files you are referring to. The exception deletions were because I wanted to re-use the admin exceptions rather than introduce new ui version. The governance endpoint deletion and the like was part of the design change in line with the way the OMAGServer.

File: open-metadata-implementation/access-services/discovery-engine/discovery-engine-server/src/main/java/org/odpi/openmetadata/accessservices/discoveryengine/handlers/DiscoveryConfigurationHandler.java (hunk 1 above)
Comment (davidradl) [reply open-metadata-implementation/access-services/discovery-engine/discovery-engine-server/src/main/java/org/odpi/openmetadata/accessservices/discoveryengine/handlers/DiscoveryConfigurationHandler.java]: I think this comment refers to an outdated file. This file is not touched in the latest code change for this PR.

**Comment by mandy-chessell** [comment]
The commit https://github.com/odpi/egeria/pull/1939/commits/d682ee37241fba2b77f8106169c8fca3e33b49ea shows part of the Discovery Engine OMAS server support being deleted.

**Comment by planetf1** [comment]
@mandy-chessell That commit seems to only refer to UI components which I'd suggested were extraneous in an earlier review?

**Comment by davidradl** [comment]
@mandy-chessell @planetf1 the latest code does not have these incorrect deletions - you can check in the file changes

File: open-metadata-implementation/common-services/multi-tenant/src/main/java/org/odpi/openmetadata/commonservices/multitenant/OCFOMVSServiceInstanceHandler.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+
+    /**
+     * Return the service's official name
+     *
+     * @param callingServiceURLName url fragment that indicates the service name
+     * @return String name
+     */
+    public String  getServiceName(String callingServiceURLName)
```
[the remark this answers is not in the document]
Comment (davidradl) [reply open-metadata-implementation/common-services/multi-tenant/src/main/java/org/odpi/openmetadata/commonservices/multitenant/OCFOMVSServiceInstanceHandler.java]: Sorry @bogdan-sava I am not sure what you are suggesting.

File: open-metadata-implementation/admin-services/admin-services-server/src/main/java/org/odpi/openmetadata/adminservices/OMAGOperationalServicesInstance.java [hunk 3]
```diff
@@ -24,7 +24,7 @@
  * OMAGOperationalServicesInstance provides the references to the active services for an instance of an OMAG Server.
  */
 
-class OMAGOperationalServicesInstance extends OMAGServerServiceInstance
+public class OMAGOperationalServicesInstance extends OMAGServerServiceInstance
 {
     private OMAGServerConfig                    operationalConfiguration            = null;
     private OMRSOperationalServices             operationalRepositoryServices       = null;
```
[the remark this answers is not in the document]
Comment (davidradl) [reply open-metadata-implementation/admin-services/admin-services-server/src/main/java/org/odpi/openmetadata/adminservices/OMAGOperationalServicesInstance.java]: Agreed - but I do not think it does any harm to be explicit

File: open-metadata-implementation/common-services/multi-tenant/src/main/java/org/odpi/openmetadata/commonservices/multitenant/OCFOMVSServiceInstanceHandler.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            callingServiceName = ViewServiceDescription.SUBJECT_AREA.getViewServiceName() + " OMVS";
+        }
+        else if (openLineageURLName.equals(callingServiceURLName))
+        {
+            callingServiceName = ViewServiceDescription.TYPE_EXPLORER.getViewServiceName() + " OMVS";
+        }
+        else if (typeExplorerURLName.equals(callingServiceURLName))
+        {
+            callingServiceName = ViewServiceDescription.TYPE_EXPLORER.getViewServiceName() + " OMVS";
```
[the remark this answers is not in the document]
Comment (davidradl) [reply open-metadata-implementation/common-services/multi-tenant/src/main/java/org/odpi/openmetadata/commonservices/multitenant/OCFOMVSServiceInstanceHandler.java]: Agreed will change

File: open-metadata-implementation/common-services/multi-tenant/src/main/java/org/odpi/openmetadata/commonservices/multitenant/ffdc/UIServerInstanceErrorCode.java [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+    /**
+     * Returns the error message with the placeholders filled out with the supplied parameters.
+     *
+     * @param params   strings that plug into the placeholders in the errorMessage
+     * @return errorMessage (formatted with supplied parameters)
+     */
+    public String getFormattedErrorMessage(String... params)
+    {
+        log.debug(String.format("<== OMAGServerInstanceErrorCode.getMessage(%s)", Arrays.toString(params)));
```
[the remark this answers is not in the document]
Comment (davidradl) [reply open-metadata-implementation/common-services/multi-tenant/src/main/java/org/odpi/openmetadata/commonservices/multitenant/ffdc/UIServerInstanceErrorCode.java]: changed

File: open-metadata-implementation/common-services/multi-tenant/src/main/java/org/odpi/openmetadata/commonservices/multitenant/OCFOMVSServiceInstanceHandler.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidradl) [reply open-metadata-implementation/common-services/multi-tenant/src/main/java/org/odpi/openmetadata/commonservices/multitenant/OCFOMVSServiceInstanceHandler.java]: This is the same logic as the equivalent OMAS class. I think we should keep it as is for consistency. Please raise a separate issue for this - if you want to change it.

**Pull request merged**